src/periphery/precrime/OFTV2View.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "interfaces/IOFTV2View.sol";
+import "tokens/LzIndirectOFTV2.sol";
+
+contract OFTV2View is IOFTV2View {
+    using BytesLib for bytes;
+
+    LzIndirectOFTV2 immutable oft;
+    IERC20 immutable token;
+    ILzEndpoint immutable endpoint;
+
+    /// @notice Local decimals to shared decimals rate
+    uint immutable ld2sdRate;
+
+    constructor(address _oft) {
+        oft = LzIndirectOFTV2(_oft);
+        token = LzIndirectOFTV2(oft).innerToken();
+        endpoint = LzOFTCoreV2(_oft).lzEndpoint();
+
+        (, bytes memory data) = ILzCommonOFT(_oft).token().staticcall(abi.encodeWithSignature("decimals()"));
```

Comment:
maybe check for length as well, like boringErc20 is doing or use BoringERC20 directly?
```
function safeDecimals(IERC20 token) internal view returns (uint8) {
  (bool success, bytes memory data) = address(token).staticcall(abi.encodeWithSelector(SIG_DECIMALS));
  return success && data.length == 32 ? abi.decode(data, (uint8)) : 18;
}
```